### ISSUE TITLE
[BUG] - Input probles

### ISSUE THREAD

**Issue by andresflorez16** [nextui-org/nextui#582]
### Describe the bug

I tried to use the Input with id, type attributes, but I got the same error

### Steps to Reproduce the Bug or Issue

1. Import Input
2. Add Attributs to my Input
3. Console Error

### Expected behavior

As I user, I expect no got Console Error

### Screenshots or Videos

![Screenshot_1](https://user-images.githubusercontent.com/88729409/176370128-5b2293e2-6fe7-439a-8fb6-49c3823cec38.png)
![Screenshot_2](https://user-images.githubusercontent.com/88729409/176370130-ca2e726a-9875-43dd-a840-93c24fa89ca5.png)

### Operating System Version

- OS: [Windows]

### Browser

Chrome

**Comment by danielsimao** [comment]
@andresflorez16, did you setup correctly the library? You need to wrapper your app around `NextUIProvider` as it is explained in the [Getting Started](https://nextui.org/docs/guide/getting-started#setup).

This erros occurs because the library `react-aria` generates unique ids but needs a specific provider to generate the same id both on the server side and the client side.

**Comment by andresflorez16** [comment]
Hi, thanks for your prompt response, I use the provider:
![image](https://user-images.githubusercontent.com/88729409/176458034-891a0bce-a394-42a6-9b63-852b97174eb7.png)
![image](https://user-images.githubusercontent.com/88729409/176458109-447bb566-e191-4c27-88f1-6f5e720e8aed.png)

**Comment by andresflorez16** [comment]
This is the repository, https://github.com/andresflorez16/ladrillos-mocoa.git

**Comment by andresflorez16** [comment]
Everything it's fine, idk what it could be

**Comment by danielsimao** [comment]
Can this be closed then?

**Comment by andresflorez16** [comment]
Ok close it, but the bug is still there, so I'll keep an eye on it

**Comment by tpruisimao** [comment]
Sorry, I misundertood. I thought you said the warning was gone.

**Comment by andresflorez16** [comment]
Oh no, sorry, the warning is still there, I mean, idk what it's, my code it's fine... 😅

**Comment by Tossko** [comment]
I have the same error as @andresflorez16, i have the provider configured according to the official documentation.

![image](https://user-images.githubusercontent.com/39286714/176559577-cd49b0e7-0666-4ca9-bf86-c731341cf2d2.png)

**Comment by danielsimao** [comment]
I just found why: [React Spectrum Issue](https://github.com/adobe/react-spectrum/issues/2231#issuecomment-905824601). So, this is not really a bug, but there is a solution.

If you are running a React version previous to 18 you will only be able to suppress that error by disabling Strict Mode. Which is not great if you wanna run your app on Strict Mode. Here is how you disable it in your Nextjs app.

> next.config.js

```
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: false,
};

module.exports = nextConfig;
```
[link to docs](https://nextjs.org/docs/api-reference/next.config.js/react-strict-mode)

You will be able to run Strict Mode on React 18 because of the new hook `useId` that will handle this SSR rendering cases.

**Comment by andresflorez16** [comment]
`/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: false,
};

module.exports = nextConfig;`

That's works for me, thanks.

**Comment by danielsimao** [comment]
This ticket can be closed now, but here are some necessary actions for future @jrgarciadev:

1. Upgrade current React version to `18.0.0`
2. Update `@react-aria`

**Comment by andresflorez16** [comment]
My versions are > 18.0.0 btw

**Comment by danielsimao** [comment]
Yes, but the problem comes from `react-aria` which is a dependency package of the project. The current version most likely does not have react in version `18.0.0`.

**Comment by jrgarciadev** [comment]
Hey @danielsimao thank you for finding a solution, this is because react-aria doesn't support the strict mode of `React 18` yet https://github.com/adobe/react-spectrum/pull/3108, Strict mode support will be done separately
